### DISCUSSION TITLE
Reusable JavaScript

### DISCUSSION THREAD

**Discussion by thomaskamalakis** [Budibase/budibase#11331] [Ideas]
Hi budibase team

It would be great if there was someway of sharing JavaScript functions functions across views. So say I have defined a JavaScript functions yadayada(param1,param2) somewhere in view /dashboard to have a way of using it in view /home without rewriting the code - sort of a DRY approach.

Thanks

**Comment by meticulousfan** [comment]
Hi, @thomaskamalakis.
I don't understand what you are referring to about "views".
Could you tell me more in detail?

**Reply by thomaskamalakis** [reply]
Hi @meticulousfan 

Sorry by "views" I meant "screens"

So say I have a form in one screen and to make a validation in a field I am using a custom JavaScript function that I wrote. If I need to use it again in another screen (or in another field of the same form for that matter) I must copy/paste its code. 

It would be great to have some way of reusing the code by putting it in a module. Say like node's `require`

Cheers

Thomas

**Comment by melohagan** [comment]
Would be nice! Open feature request here: https://github.com/Budibase/budibase/issues/3571
